File: C++/2.EvenFibonachi.cpp
#include <iostream>
#include <vector>

// Function to generate the Fibonacci sequence up to n elements
std::vector<int> generateFibonacci(int n) {
    std::vector<int> fibonacciSequence = {0, 1};

    while (fibonacciSequence.size() < n) {
        int nextNumber = fibonacciSequence.back() + fibonacciSequence[fibonacciSequence.size() - 2];
        fibonacciSequence.push_back(nextNumber);
    }

    return fibonacciSequence;
}

// Function to find the sum of even Fibonacci numbers under a given limit
int sumEvenFibonacci(int limit) {
    // Call the generateFibonacci function and give the limit to it
    std::vector<int> fibonacciNumbers = generateFibonacci(34); // Assuming 34 elements are enough for the example

    // Check if the generated number is divisible by two and sum the even Fibonacci numbers under the limit
    int evenSum = 0;
    for (int num : fibonacciNumbers) {
        if (num < limit && num % 2 == 0) {
            evenSum += num;
        }
    }

    return evenSum;
}

int main() {
    // Example: Find the sum of even Fibonacci numbers under 4 million
    int result2 = sumEvenFibonacci(4000000);
    std::cout << result2 << std::endl;

    return 0;
}

File: Pyhton/2.EvenFibonachi.py

def generate_fibonacci(n):
    
    fibonacci_sequence = [0, 1]

    while len(fibonacci_sequence) < n:
        next_number = fibonacci_sequence[-1] + fibonacci_sequence[-2]
        fibonacci_sequence.append(next_number)

    return fibonacci_sequence

def sum_even_fibonacci(limit):
    #call the our fibonacci function and give the limit to it 
    fibonacci_numbers = generate_fibonacci(34) 
    #check if the genarated number dividble on two or not 
    even_sum = sum(num for num in fibonacci_numbers if num < limit and num % 2 == 0)
    return even_sum

# Example: Find the sum of even Fibonacci numbers under 4 million
result2 = sum_even_fibonacci(4000000)
print(result2)
